### PR TITLE
Let TestCase contain all data currently read from TestSuite and put it as a field in TestCaseRunner

### PR DESCRIPTION
**Pull request by mkaput**

### CODE REVIEW

File: protostar/commands/test/test_case_runners/fuzz_test_case_runner.py [hunk 1]
```diff
@@ -8,27 +8,27 @@
...
     FailedFuzzTestCaseResult,
     FailedTestCaseResult,
     FuzzResult,
     PassedFuzzTestCaseResult,
     TestResult,
 )
-from protostar.commands.test.test_environment_exceptions import ReportedException
 
 
 class FuzzTestCaseRunner(TestCaseRunner[FuzzTestExecutionResult]):
```
Comment (mkaput) [inline protostar/commands/test/test_case_runners/fuzz_test_case_runner.py:21]: Note: the subclasses of `TestCaseRunner` are going to be removed here: https://github.com/software-mansion/protostar/commit/8b284ef3ba4009938880e7f58b0e99f645a75580

File: protostar/commands/test/test_case_runners/fuzz_test_case_runner.py [hunk 2]
```diff
@@ -8,27 +8,27 @@
...
     def __init__(
-        self,
-        fuzz_test_execution_environment: FuzzTestExecutionEnvironment,
-        test_case_runner_deps: TestCaseRunner.Dependencies,
+        self, fuzz_test_execution_environment: FuzzTestExecutionEnvironment, **kwargs
     ) -> None:
-        super().__init__(test_case_runner_deps)
+        super().__init__(**kwargs)
```
Comment (kasperski95) [inline protostar/commands/test/test_case_runners/fuzz_test_case_runner.py:25]: typechecker can't verify kwargs

File: protostar/commands/test/test_case_runners/test_case_runner.py [hunk 3]
```diff
@@ -6,63 +6,56 @@
...
-        test_case_name: str
         execution_time: float
 
-    def __init__(self, dependencies: Dependencies) -> None:
-        super().__init__()
-        self._test_suite = dependencies.test_suite
-        self._output_recorder = dependencies.output_recorder
+    def __init__(self, test_case: TestCase, output_recorder: OutputRecorder) -> None:
+        self.test_case = test_case
+        self.output_recorder = output_recorder
```
Comment (kasperski95) [inline protostar/commands/test/test_case_runners/test_case_runner.py:28]: Shouldn't be these props private?

File: protostar/commands/test/test_suite.py [hunk 4]
```diff
@@ -5,6 +5,7 @@
 
 @dataclass(frozen=True)
 class TestCase:
+    test_path: Path
     test_fn_name: str
     setup_fn_name: Optional[str] = None
```
Comment (kasperski95) [inline protostar/commands/test/test_suite.py:10]: Maybe `TestCase` should have a reference to test suite instead of `test_path`.

**Review by kasperski95 — APPROVED** [review]

File: protostar/commands/test/test_case_runners/test_case_runner.py (hunk 3 above)
Comment (mkaput) [reply protostar/commands/test/test_case_runners/test_case_runner.py:28]: Done.

**Comment by mkaput** [comment]
Didn't think about this :+1: Done.

File: protostar/commands/test/test_suite.py (hunk 4 above)
Comment (mkaput) [reply protostar/commands/test/test_suite.py:10]: Didn't think about this :+1: Done.

File: protostar/commands/test/test_case_runners/fuzz_test_case_runner.py (hunk 2 above)
Comment (kasperski95) [reply protostar/commands/test/test_case_runners/fuzz_test_case_runner.py:25]: I've got a feeling I'm going to complain in the next following PR. Anyway, resolving.

**Review by kasperski95 — APPROVED** [review]

**Pull request merged**